numba/tests/test_parallel_backend.py

```diff
@@ -954,6 +955,91 @@ def forker():
             print("OUT:", out)
             print("ERR:", err)
 
+    @needs_external_compilers
+    @linux_only # os.fork required.
+    def test_lifetime_of_task_scheduler_handle(self):
+        # See PR #7280 for context.
+        BROKEN_COMPILERS = 'SKIP: COMPILATION FAILED'
+        runme = """if 1:
+            import ctypes
+            import os
+            import sys
+            from tempfile import TemporaryDirectory, NamedTemporaryFile
+            from numba.pycc.platform import Toolchain, _external_compiler_ok
+            from numba import njit, prange, threading_layer
+
+            if not _external_compiler_ok:
+                raise AssertionError('External compilers are not found.')
+
+            try:
+                with TemporaryDirectory() as tmpdir:
+                    with NamedTemporaryFile(dir=tmpdir) as tmpfile:
+                        src = \"\"\"
+                        #define TBB_PREVIEW_WAITING_FOR_WORKERS 1
+                        #include <tbb/tbb.h>
+                        static tbb::task_scheduler_handle tsh;
+                        extern "C"
+                        {
+                        void launch(void)
+                        {
+                            tsh = tbb::task_scheduler_handle::get();
+                        }
+                        }
+                        \"\"\"
+                        cxxfile = f"{tmpfile.name}.cxx"
+                        with open(cxxfile, 'wt') as f:
+                            f.write(src)
+                        tc = Toolchain()
+                        object_files = tc.compile_objects([cxxfile,],
+                                                           output_dir=tmpdir)
+                        dso_name = f"{tmpfile.name}.so"
+                        tc.link_shared(dso_name, object_files,
+                                       libraries=['tbb',],
+                                       export_symbols=['launch'])
+                        # Load into the process, it doesn't matter whether the
+                        # DSO exists on disk once it's loaded in.
+                        DLL = ctypes.CDLL(dso_name)
+            except Exception as e:
+                # Something is broken in compilation, could be one of many
+                # things including, but not limited to: missing tbb headers,
+                # incorrect permissions, compilers that don't work for the above
+                print(e)
+                print('BROKEN_COMPILERS')
+                sys.exit(0)
```

Comment:
should this return a different exit code?

Reply:
The `self.run_cmd` checks for exit code 0 and raises assertion error if it is not. What I was attempting to do here is differentiate between "This test cannot be run because of some compiler setup problem" and "The compiler setup was fine and the test can be run, but is has failed". The test uses a non-zero exit code which results in the assertion error to signal the latter and a zero exit code with a special token assigned to the `BROKEN_COMPILERS` variable for the former. Should the `BROKEN_COMPILERS` condition occur, the test sets a `self.skipTest()`.

Reply:
As agreed in OOB conversion, there are too many ways for the compilation to fail. We will allow the test to be a little fragile (may accidental skip). Later we will look into ways to make sure the test is not skipped on systems that we have tight control and can guarantee the test must work.